src/blocks/CMakeLists.txt

```diff
@@ -26,14 +26,20 @@
 # STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
 # THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-if(APPLE)
-    add_library(blocks STATIC blockexports.c)
-    set_target_properties(blocks PROPERTIES LINKER_LANGUAGE C)
-else()
-    add_custom_command(OUTPUT blocks.o MAIN_DEPENDENCY blocks.dat COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR} && ${CMAKE_LINKER} ${LD_RAW_FLAGS} -r -b binary -o ${CMAKE_CURRENT_BINARY_DIR}/blocks.o blocks.dat)
-    add_custom_command(OUTPUT testnet_blocks.o MAIN_DEPENDENCY testnet_blocks.dat COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR} && ${CMAKE_LINKER} ${LD_RAW_FLAGS} -r -b binary -o ${CMAKE_CURRENT_BINARY_DIR}/testnet_blocks.o testnet_blocks.dat)
-    add_custom_command(OUTPUT stagenet_blocks.o MAIN_DEPENDENCY stagenet_blocks.dat COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR} && ${CMAKE_LINKER} ${LD_RAW_FLAGS} -r -b binary -o ${CMAKE_CURRENT_BINARY_DIR}/stagenet_blocks.o stagenet_blocks.dat)
-    add_library(blocks STATIC blocks.o testnet_blocks.o stagenet_blocks.o blockexports.c)
-    set_target_properties(blocks PROPERTIES LINKER_LANGUAGE C)
-endif()
+set(GENERATED_SOURCES "")
 
+foreach(BLOB_NAME checkpoints testnet_blocks stagenet_blocks)
+    set(OUTPUT_C_SOURCE "generated_${BLOB_NAME}.c")
+    list(APPEND GENERATED_SOURCES ${OUTPUT_C_SOURCE})
+    set(INPUT_DAT_FILE "${BLOB_NAME}.dat")
+    set(GENERATE_C_SOURCE_FROM_BLOB perl -e "\"local$$/\;@blob=unpack('C*',<>)\;print('const\tunsigned\tchar\t${BLOB_NAME}[]={',join(',',@blob),'}\;\tconst\tunsigned\tint\t${BLOB_NAME}_len=',0+@blob,'\;')\"")
```

Comment:
Why the change from size_t to unsigned int ?

Reply:
Just to keep the perl script as simple as possible, to omit `#include <stddef.h>` line.

Reply:
Updated, added the include to generated files to use `size_t`